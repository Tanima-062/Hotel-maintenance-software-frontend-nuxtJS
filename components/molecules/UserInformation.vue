<template>
  <div>
    <div
      class="fixed w-screen h-screen top-0 left-0 z-40"
      @click.self="closeUserInfo"
    ></div>
    <div
      class="userinfo-bg shadow border border-solid border-white absolute right-0 p-3 bg-white mt-2 rounded z-50"
    >
      <div class="mb-3">
        <p class="text-13px font-extralight text-black-light">
          {{ $t('global.header.userInfo.lastName') }}
        </p>
        <p class="font-normal leading-5 text-black">
          {{ userInfo.lastname }}
        </p>
      </div>
      <div class="mb-3">
        <p class="text-13px ont-extralight text-black-light">
          {{ $t('global.header.userInfo.firstName') }}
        </p>
        <p class="font-normal leading-5 text-black">
          {{ userInfo.firstname }}
        </p>
      </div>
      <div>
        <p class="text-13px text-black-light">
          {{ $t('global.header.userInfo.email') }}
        </p>
        <p class="font-normal leading-5 text-black break-words">
          {{ userInfo.email }}
        </p>
      </div>
    </div>
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'vue-property-decorator'
import { Account } from '~/types'

@Component
export default class extends Vue {
  @Prop({ required: true })
  userInfo!: Account.UserInformation

  @Prop({ required: true })
  closeUserInfo!: Function
}
</script>
<style lang="sass" scoped>
.userinfo-bg
  min-width: 230px
  max-width: 340px
</style>
