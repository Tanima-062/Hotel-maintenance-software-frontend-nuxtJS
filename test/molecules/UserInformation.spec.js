import { mount } from '@vue/test-utils'
import Vue from 'vue'
import VueI18n from 'vue-i18n'
import UserInformation from '@/components/molecules/UserInformation.vue'

Vue.use(VueI18n)
describe('Molecules UserInformation Check', () => {
  test('UserInformation Test', () => {
    const props = {
      userInfo: {
        firstname: 'test',
        lastname: 'test',
        email: 'test',
      },
      closeUserInfo: () => {},
    }
    const wrapper = mount(UserInformation, {
      propsData: props,
      mocks: {
        $t: (msg) => msg,
      },
    })
    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
