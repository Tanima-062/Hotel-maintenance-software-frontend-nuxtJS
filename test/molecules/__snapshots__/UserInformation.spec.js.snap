// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules UserInformation Check UserInformation Test 1`] = `
<div>
  <div
    class="fixed w-screen h-screen top-0 left-0 z-40"
  />
   
  <div
    class="userinfo-bg shadow border border-solid border-white absolute right-0 p-3 bg-white mt-2 rounded z-50"
  >
    <div
      class="mb-3"
    >
      <p
        class="text-13px font-extralight text-black-light"
      >
        
        global.header.userInfo.lastName
      
      </p>
       
      <p
        class="font-normal leading-5 text-black"
      >
        
        test
      
      </p>
    </div>
     
    <div
      class="mb-3"
    >
      <p
        class="text-13px ont-extralight text-black-light"
      >
        
        global.header.userInfo.firstName
      
      </p>
       
      <p
        class="font-normal leading-5 text-black"
      >
        
        test
      
      </p>
    </div>
     
    <div>
      <p
        class="text-13px text-black-light"
      >
        
        global.header.userInfo.email
      
      </p>
       
      <p
        class="font-normal leading-5 text-black break-words"
      >
        
        test
      
      </p>
    </div>
  </div>
</div>
`;
